server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -112,21 +125,45 @@ public final class IndicesRequestCache implements RemovalListener<IndicesRequest
         new TimeValue(0),
         Property.NodeScope
     );
+    public static final Setting<TimeValue> INDICES_REQUEST_CACHE_CLEAN_INTERVAL_SETTING = Setting.positiveTimeSetting(
+        "indices.requests.cache.cleanup.interval",
+        TimeValue.timeValueMinutes(1),
+        Property.NodeScope
+    );
+    public static final Setting<String> INDICES_REQUEST_CACHE_STALENESS_THRESHOLD_SETTING = new Setting<>(
+        "indices.requests.cache.cleanup.staleness_threshold",
+        "0%",
+        IndicesRequestCache::validateStalenessSetting,
+        Property.NodeScope
```

Comment:
I intentionally wanted this to be static. 
In a scenario where the user changes the setting from 0 to higher, we will only then start calculating the staleness. This means any staleness build up after the previous regular cleanup cycle would go unaccounted for which would result in an in-accurate staleness count. This could be mainly in a scenario where the users changed from 0 to something very high, lets say 90%, may be the unaccounted staleness was greater than 10% and we might end up evicting from cache.

Reply:
Can we use `keysToClean` information to populate the map as part of settings update ?

Reply:
We are breaking backward compatibility with older setting unable to control the cache cleaner time interval now. We need to set the default value of this new setting to whatever existing setting value is.

Reply:
>We are breaking backward compatibility with older setting unable to control the cache cleaner time interval now. We need to set the default value of this new setting to whatever existing setting value is.

Good point, I didn't think of this. Addressed.

Reply:
>Can we use keysToClean information to populate the map as part of settings update ? We may need to think about race conditions if any though

I didn't understand.

Reply:
>Can we use keysToClean information to populate the map as part of settings update ? 

I don't think we can do this as it doesn't have complete info. We will need to track desired keys(for a shardId and readerCacheKeyId combination) from the start via cacheCleanupManager.updateCleanupKeyToCountMapOnCacheInsertion(cleanupKey).

Though I agree of having staleness setting to be a dynamic one.